vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/pool/SqlConnectionPool.java

```diff
@@ -309,7 +313,8 @@ public class PooledConnection implements Connection, Connection.Holder {
       this.factory = factory;
       this.conn = conn;
       this.listener = listener;
-      this.lifetimeEvictionTimestamp = maxLifetime > 0 ? System.currentTimeMillis() + maxLifetime : Long.MAX_VALUE;
+      long randomizedJitter = (maxLifetime > 0 && jitter > 0 && maxLifetime > jitter) ? ThreadLocalRandom.current().nextLong(-jitter, jitter + 1) : 0;
```

Comment:
maxLifetime > 0 is not needed since the following line will ignore maxLifetime and jitter both if maxLifetime is not defined.

Reply:
maxLifetime > jitter this should also be removed. It should not be mandated that jitter is always less than maxLifetime.